iroha/src/block_sync.rs

```diff
@@ -195,9 +202,10 @@ pub mod message {
                                 }
                             }
                         }
-                    } else {
-                        log::error!("Error: not sending any blocks as batch_size is equal to zero.")
-                    }
+                        None => log::error!(
+                            "Error: not sending any blocks as batch_size is equal to zero."
+                        ),
```

Comment:
Also should be `warn` to my mind, also please change the error message. Here the message should be that there are no blocks after `hash`.